workflow-ui/compose/src/main/java/com/squareup/workflow1/ui/compose/ComposeRendering.kt

```diff
@@ -0,0 +1,63 @@
+package com.squareup.workflow1.ui.compose
+
+import androidx.compose.runtime.Composable
+import com.squareup.workflow1.ui.AndroidViewRendering
+import com.squareup.workflow1.ui.ViewEnvironment
+import com.squareup.workflow1.ui.ViewFactory
+import com.squareup.workflow1.ui.ViewRegistry
+import com.squareup.workflow1.ui.WorkflowUiExperimentalApi
+import kotlin.reflect.KClass
+
+/**
+ * Interface implemented by a rendering class to allow it to drive a composable UI via an
+ * appropriate [ComposeViewFactory] implementation. This is the compose analog to
+ * [AndroidViewRendering].
+ *
+ * ```
+ * @OptIn(WorkflowUiExperimentalApi::class)
+ * data class HelloView(
+ *   val message: String,
+ *   val onClick: () -> Unit
+ * ) : ComposeRendering<HelloView> {
+ *
+ *   @Composable override fun Content(viewEnvironment: ViewEnvironment) {
+ *     Button(onClick) {
+ *       Text(message)
+ *     }
+ *   }
+ * }
+ * ```
+ *
+ * This is the simplest way to bridge the gap between your workflows and the UI, but using it
+ * requires your workflows code to reside in Android modules, instead of pure Kotlin. If this is a
+ * problem, or you need more flexibility for any other reason, you can use [ViewRegistry] to bind
+ * your renderings to [ComposeViewFactory] implementations at runtime.
+ */
+@WorkflowUiExperimentalApi
+public interface ComposeRendering<V : ComposeRendering<V>> : AndroidViewRendering<V> {
+
+  /** Don't override this, override [Content] instead. */
+  override val viewFactory: ViewFactory<V> get() = Companion
```

Comment:
I'm torn between making this a class so I can make this property `final override`, or keeping it an interface to keep it consistent with `AndroidViewRendering`. WDYT?

Reply:
`abstract class` makes sense to me if you want to enforce this.

Reply:
@steve-the-edwards votes for a class.
@rjrjr votes for… both? 😛

Reply:
FTR, the only really material argument I could think of for keeping it an interface was that renderings might be defined as sealed classes, and if only some cases would want to implement this, they'd need it to be an interface. But I think we actually discourage sealed classes for renderings, and even if someone wants to do that we have sealed interfaces now so it's not a problem.